### PR TITLE
Docs

### LINKED ISSUE #1326: Known issues documentation
Add documentation (maybe like FAQ) about known issues:
1. ~AllowSynchronousIO: #1116, #1299, #1325, graphql-dotnet/server#271, #1317, graphql-dotnet/server#277~
2. ~DI: Scoped services issue: #1345, #1160 and related issues~
3. ~Auto generating classes and other stuff #576~
4. ~Auto registering output and input types #897~
5. Using derived graph types and cyclic dependencies #1010  -   **too narrow to document, cancelled**
6. ~File downloading #511~
7. ~Note about two authorization models - generic and ASP.NET Core graphql-dotnet/authorization#68 graphql-dotnet/authorization#74~
8. ~Note about descriptions inheritance #1004~ 
9. ~Recommendations to always use singleton schema and graph types over scoped ones #1011~
10. ~Many to many with dataLoader #1625~j
11. ~Subscription Authentication #1244~
12. ~Flags enums #1666~
13. ~`null` in scalars #1542~

### PR DESCRIPTION
**Pull request by sungam3r**
Closes #1326 

> 5. Using derived graph types and cyclic dependencies

I thought it was too narrow topic and did not burden the documentation with it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,10 +82,13 @@ It supports the popular IDEs for managing GraphQL requests and exploring GraphQL
 
 ## Upgrade Guides
 
-* 2.4.0 to 3.0 - under development
+* [3.x to 4.x](https://graphql-dotnet.github.io/docs/guides/migration4)
```
Comment (sungam3r) [inline README.md:85]: Let it be here even page does not exist yet.

Comment (Shane32) [reply README.md:85]:
```suggestion
* [3.x to 4.x](https://github.com/graphql-dotnet/graphql-dotnet/blob/develop/docs2/site/docs/guides/migration4.md)
```

Comment (Shane32) [reply README.md:85]: Let's make the link actually work at least.

Comment (sungam3r) [reply README.md]: OK I didn't think about it.

**Review by Shane32 — APPROVED** [review]

**Review by joemcbride — APPROVED** [review]

**Pull request merged**